src/pydvl/utils/score.py

```diff
@@ -0,0 +1,126 @@
+"""
+This module provides a :class:`Scorer` class that wraps scoring functions with
+additional information.
+
+Scorers can be constructed in the same way as in scikit-learn: either from 
+known strings or from a callable. Greater values must be better. If they are not,
+a negated version can be used, see scikit-learn's `make_scorer()
+<https://scikit-learn.org/stable/modules/generated/sklearn.metrics.make_scorer.html>`_.
+
+:class:`Scorer` provides additional information about the scoring function, like
+its range and default values.
+"""
+from typing import Callable, Optional, Protocol, Tuple, Union
+
+import numpy as np
+from numpy.typing import NDArray
+from scipy.special import expit
+from sklearn.metrics import get_scorer
+
+from pydvl.utils.types import SupervisedModel
+
+__all__ = ["Scorer", "compose_score", "squashed_r2", "squashed_variance"]
+
+
+class ScorerCallable(Protocol):
+    """Signature for a scorer"""
+
+    def __call__(self, model: SupervisedModel, X: NDArray, y: NDArray) -> float:
+        ...
+
+
+class Scorer:
+    """A scoring callable that takes a model, data, and labels and returns a
+    scalar.
+
+    :param scoring: Either a string or callable that can be passed to
+        `get_scorer
+        <https://scikit-learn.org/stable/modules/generated/sklearn.metrics.get_scorer.html>`_.
+    :param default: score to be used when a model cannot be fit, e.g. when too
+        little data is passed, or errors arise.
+    :param range: numerical range of the score function. Some Monte Carlo
+        methods can use this to estimate the number of samples required for a
+        certain quality of approximation. If not provided, it can be read from
+        the ``scoring`` object if it provides it, for instance if it was
+        constructed with :func:`~pydvl.utils.types.compose_score`.
+    :param name: The name of the scorer. If not provided, the name of the
+        function passed will be used.
+
+    .. versionadded:: 0.5.0
+
+    """
+
+    _name: str
+    range: NDArray[np.float_]
+
+    def __init__(
+        self,
+        scoring: Union[str, ScorerCallable],
+        default: float = np.nan,
+        range: Tuple = (-np.inf, np.inf),
+        name: Optional[str] = None,
+    ):
+        self._scorer = get_scorer(scoring)
+        self.default = default
+        # TODO: auto-fill from known scorers ?
```

Comment:
Should we just hard-code it?

We can get the names of the scorers from [here](https://github.com/scikit-learn/scikit-learn/blob/8c9c1f27b/sklearn/metrics/_scorer.py#L842).

I will create an issue for this but since it's only used for group testing it doesn't really have a high priority.